homeassistant/components/rainmachine/switch.py

```diff
@@ -171,12 +186,28 @@ async def async_setup_entry(
     for uid, program in programs_coordinator.data.items():
         entities.append(
             RainMachineProgram(
-                programs_coordinator, controller, uid, program["name"], entry
+                programs_coordinator,
+                controller,
+                entry,
+                RainMachineSwitchDescription(
+                    key=f"RainMachineProgram_{uid}",
+                    name=program["name"],
+                    uid=uid,
```

Comment:
Ideally the descriptions would be populated with static contend, but I guess it will work.

Reply:
The names/IDs come from dynamic data, so I don't think we can do that here.